components/places/src/db/schema.rs

```diff
@@ -74,20 +76,34 @@ const CREATE_TABLE_INPUTHISTORY_SQL: &str = "CREATE TABLE moz_inputhistory (
 // XXX - TODO - moz_annos
 // XXX - TODO - moz_anno_attributes
 // XXX - TODO - moz_items_annos
-// XXX - TODO - moz_bookmarks
-// XXX - TODO - moz_bookmarks_deleted
 
-// TODO: This isn't the complete `moz_bookmarks` definition, just enough to
-// test autocomplete.
-const CREATE_TABLE_BOOKMARKS_SQL: &str = "CREATE TABLE moz_bookmarks (
+const CREATE_TABLE_BOOKMARKS_SQL: &str = r#"CREATE TABLE moz_bookmarks (
         id INTEGER PRIMARY KEY,
-        fk INTEGER,
-        title TEXT,
+        fk INTEGER DEFAULT NULL, -- place_id
+        type INTEGER NOT NULL CHECK(type BETWEEN 1 AND 3),
+        parent INTEGER,
+        position INTEGER NOT NULL CHECK(position >= 0),
+        title TEXT, -- a'la bug 1356159, NULL is special here - it means 'not edited'
+        dateAdded INTEGER NOT NULL DEFAULT 0,
         lastModified INTEGER NOT NULL DEFAULT 0,
+        guid TEXT NOT NULL UNIQUE CHECK(length(guid) == 12),
+
+        syncStatus INTEGER NOT NULL DEFAULT 0 CHECK(syncStatus BETWEEN 0 AND 2),
```

Comment:
How easy will it be for us to migrate to allow sync status == 3 here? (for Unknown)? is there a way to to change column constraints in alter table?

Reply:
> How easy will it be for us to migrate to allow sync status == 3 here? (for Unknown)?

SyncStatus::Unknown is zero.

> is there a way to to change column constraints in alter table?

Apparently not - iiuc, you need to create a new "temp" table with the correct constraints when select the entire old table into it.

Reply:
After discussions, we've removed some of these.